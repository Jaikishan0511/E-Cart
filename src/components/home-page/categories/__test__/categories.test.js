import React from "react";
import Categories from "../index"
import ReactTestRenderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
const CategoriesComp = (
    <Router>
      <Categories />
    </Router>
  );
describe("Actions", () => {
    describe("Render", () => {
      it("should be a valid React Element", () => {
        expect(React.isValidElement(<Categories />)).toBeTruthy();
      });
  
      it("should render without throwing an error", () => {
        expect(() =>
          ReactTestRenderer.create(CategoriesComp)
        ).not.toThrow();
      });
    });
  
    describe("Snapshots", () => {
      it("matches the default snapshot", () => {
        const tree = ReactTestRenderer.create(CategoriesComp).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });