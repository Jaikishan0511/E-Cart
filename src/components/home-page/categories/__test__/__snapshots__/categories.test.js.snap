// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions Snapshots matches the default snapshot 1`] = `
<div
  className="multiple-categories"
>
  <div
    className="close-category-icon"
    onClick={[Function]}
  >
    <img
      alt="close-icon"
      src="/images/close-icon.png"
    />
  </div>
  <div
    className="categories-label"
  >
    <input
      defaultChecked={true}
      id={0}
      name="formHorizontalRadios"
      onClick={[Function]}
      type="radio"
      value="All"
    />
    <label
      className="radio-lable"
    >
      All
    </label>
  </div>
  <div
    className="categories-label"
  >
    <input
      defaultChecked={false}
      id={1}
      name="formHorizontalRadios"
      onClick={[Function]}
      type="radio"
      value="Electronics"
    />
    <label
      className="radio-lable"
    >
      Electronics
    </label>
  </div>
  <div
    className="categories-label"
  >
    <input
      defaultChecked={false}
      id={2}
      name="formHorizontalRadios"
      onClick={[Function]}
      type="radio"
      value="Food Essentials"
    />
    <label
      className="radio-lable"
    >
      Food Essentials
    </label>
  </div>
  <div
    className="categories-label"
  >
    <input
      defaultChecked={false}
      id={3}
      name="formHorizontalRadios"
      onClick={[Function]}
      type="radio"
      value="Home & furniture"
    />
    <label
      className="radio-lable"
    >
      Home & furniture
    </label>
  </div>
  <div
    className="categories-label"
  >
    <input
      defaultChecked={false}
      id={4}
      name="formHorizontalRadios"
      onClick={[Function]}
      type="radio"
      value="Fashion"
    />
    <label
      className="radio-lable"
    >
      Fashion
    </label>
  </div>
  <div
    className="categories-label"
  >
    <input
      defaultChecked={false}
      id={5}
      name="formHorizontalRadios"
      onClick={[Function]}
      type="radio"
      value="Sports"
    />
    <label
      className="radio-lable"
    >
      Sports
    </label>
  </div>
</div>
`;
